LiteCore/Database/DatabaseImpl.cc

```diff
@@ -767,6 +771,18 @@ namespace litecore {
     }
 
 
+    void DatabaseImpl::collectionRemoved(slice scope, slice name) {
+        LOCK(_collectionsMutex);
+        CollectionSpec spec {name, scope};
+        if (auto c = _collections.find(spec); c != _collections.end()) {
+            asInternal(c->second.get())->close();
```

Comment:
This is the exact problem that I was warned against by @snej  and so you've basically just recreated the first iteration of my PR complete with the problems it comes with in terms of threading violations.